teuthology/task/tests/test_fetch_coredumps.py

```diff
@@ -0,0 +1,85 @@
+from teuthology.task.internal import fetch_binaries_for_coredumps
+from unittest.mock import patch
+import pytest
+
+
+class TestFetchCoreDumps(object):
+    class MockDecode(object):
+        def __init__(self, ret):
+            self.ret = ret
+            pass
+
+        def decode(self):
+            return self.ret
+
+    class MockPopen(object):
+        def __init__(self, ret):
+            self.ret = ret
+
+        def communicate(self, input=None):
+            return [TestFetchCoreDumps.MockDecode(self.ret)]
+
+    class MockRemote(object):
+        def __init__(self, system_type=None):
+            self.system_type = system_type
+
+        def sh(self, input=None):
+            return ""
+        
+        def _sftp_get_file(self, path=None, input=None):
+            pass
+
+        def get_file(self, path=None, input=None):
+            pass
+
+    class MockPath(object):
+        def __init__(self, ret):
+            self.ret = ret
+
+        def join(self, path=None, input=None):
+            return self.ret
+        
+        @property
+        def sep(self, path=None, input=None):
+            return self.ret
+        
+        def isdir(self, input=None):
+            return True
+        
+        def dirname(self, input=None):
+            return self.ret
+
+        def exists(self, input=None):
+            pass
+        
+    def setup(self):
+        self.the_function = fetch_binaries_for_coredumps
+
+    @patch('teuthology.task.tests.test_fetch_coredumps.TestFetchCoreDumps.MockRemote')
+    @patch('teuthology.task.internal.subprocess')
+    @patch('teuthology.task.internal.os')
+    def test_correct_format(self, m_os, m_subprocess, m_remote):
+        example_output = [
+            "1422917770.7450.core: ELF 64-bit"
+            "LSB core file x86-64, version 1 (SYSV), SVR4-style"
+            "from 'radosgw --rgw-socket-path /home/ubuntu/cephtest/"
+            "apache/tmp.client.0/fastcgi_soc'"
+        ]
+        m_os.path = TestFetchCoreDumps.MockPath("/")
+        m_os.listdir.return_value = example_output
+        m_subprocess.Popen.return_value = TestFetchCoreDumps.MockPopen(example_output[0])
+        self.the_function(None, m_remote)
+        assert m_remote.get_file.called
+    
+    @patch('teuthology.task.tests.test_fetch_coredumps.TestFetchCoreDumps.MockRemote')
+    @patch('teuthology.task.internal.subprocess')
+    @patch('teuthology.task.internal.os')
+    def test_correct_format(self, m_os, m_subprocess, m_remote):
+        example_output = [
+            "01010101010101001010101010101010110101011"
```

Comment:
Didn't notice the tests until after I made the previous comment about logging - thanks for adding these!!

Is that a realistic non-matching output? If so, I guess we might not want to log it after all...

Reply:
no haha, realistic ones are shown in the mentioned tracker. Will change it to the tracker one